src/taco-cli/cli/utils/kitComponentCommand.ts

```diff
@@ -0,0 +1,303 @@
+﻿/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../../../typings/node.d.ts" />
+/// <reference path="../../../typings/Q.d.ts" />
+/// <reference path="../../../typings/semver.d.ts" />
+/// <reference path="../../../typings/tacoKits.d.ts" />
+
+"use strict";
+
+import path = require ("path");
+import Q = require ("q");
+import semver = require ("semver");
+import util = require ("util");
+
+import CliTelemetryHelper = require ("./cliTelemetryHelper");
+import kitHelper = require ("./kitHelper");
+import resources = require ("../../resources/resourceManager");
+import TacoErrorCodes = require ("../tacoErrorCodes");
+import errorHelper = require ("../tacoErrorHelper");
+import tacoUtility = require ("taco-utils");
+
+import commands = tacoUtility.Commands;
+import CordovaHelper = tacoUtility.CordovaHelper;
+import CordovaWrapper = tacoUtility.CordovaWrapper;
+import logger = tacoUtility.Logger;
+import packageLoader = tacoUtility.TacoPackageLoader;
+import ProjectHelper = tacoUtility.ProjectHelper;
+import PromiseUtils = tacoUtility.PromisesUtils;
+import telemetryHelper = tacoUtility.TelemetryHelper;
+import utils = tacoUtility.UtilHelper;
+
+import ICommandTelemetryProperties = tacoUtility.ICommandTelemetryProperties;
+
+
+interface IKitComponentCommandDataBase extends commands.ICommandData {
+    subCommand: string,
+    targets: string[];
+};
+
+/**
+ * IKitComponentCommandData
+ *
+ * commandOptions for a kit component command
+ */
+export interface IKitComponentCommandData<T> extends IKitComponentCommandDataBase {
+    commandOptions: T;
+};
+
+/**
+ * IKitComponentInfo
+ *
+ * Represent the info for a kit component target which can be passed on to cordova
+ */
+export interface IKitComponentInfo {
+    name?: string;
+    spec?: string;
+    targetId: string;
+}
+
+/**
+ * KitComponentCommand
+ *
+ * Base class to handle add/remove operations for cordova components (platforms, plugins)
+ */
+export class KitComponentCommand extends commands.TacoCommandBase {
```

Comment:
Maybe CordovaComponentCommand? KitComponentCommand is a bit misleading given that this has code path for CLI projects as well.